boot.php

```diff
@@ -785,60 +785,85 @@ function get_scheme() {
 		return($this->scheme);
 	}
 
+	/**
+	 * @brief Retrieves the Friendica instance base URL
+	 *
+	 * Caches both SSL and non-SSL version for performance
+	 *
+	 * @param bool $ssl
+	 * @return string
+	 */
 	function get_baseurl($ssl = false) {
 
 		// Is the function called statically?
-		if (!is_object($this))
-			return(self::$a->get_baseurl($ssl));
+		if (!is_object($this)) {
+			return self::$a->get_baseurl($ssl);
+		}
 
-		$scheme = $this->scheme;
+		if (!isset($this->baseurl[$ssl ? 'https' : 'http'])) {
```

Comment:
In either way it is misleading. Because of this I suggested some comments. Like you said, even ```$ssl = false``` can produce a ````https://```` so naming this into an array key ````http```` is a little bit confusing.

(Mostly) every time when I'm coding something that could irritate others (or me, when I look at it in some months) then I add some comments to explain the background.

If you don't want to add the comment by yourself, I'm fine with it. Then I guess I will add a comment as a secondary PR to it.